src/composables/useDocumentFullscreen.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { createSharedComposable } from '@vueuse/core'
+import { readonly, ref, onBeforeMount, onBeforeUnmount } from 'vue'
+import type { Ref, DeepReadonly } from 'vue'
+
+/**
+ * Composable to check whether the page is displayed at fullscreen
+ * @return {DeepReadonly<Ref<boolean>>} - computed boolean whether the page is displayed at fullscreen
+ */
+function useDocumentFullscreenComposable() {
+	const isFullscreen = ref<boolean>(document.fullscreenElement !== null)
+
+	const changeIsFullscreen = () => {
+		isFullscreen.value = document.fullscreenElement !== null
+	}
+
+	onBeforeMount(() => {
+		document.addEventListener('fullscreenchange', changeIsFullscreen)
+		document.addEventListener('mozfullscreenchange', changeIsFullscreen)
+		document.addEventListener('MSFullscreenChange', changeIsFullscreen)
+		document.addEventListener('webkitfullscreenchange', changeIsFullscreen)
+	})
+
+	onBeforeUnmount(() => {
+		document.removeEventListener('fullscreenchange', changeIsFullscreen)
+		document.removeEventListener('mozfullscreenchange', changeIsFullscreen)
+		document.removeEventListener('MSFullscreenChange', changeIsFullscreen)
+		document.removeEventListener('webkitfullscreenchange', changeIsFullscreen)
+	})
+
+	return readonly(isFullscreen)
+}
+
+/**
+ * Enable a fullscreen with Fullscreen API
+ */
+export async function enableFullscreen() {
+	const element = document.getElementById('content-vue')
+
+	if (element.requestFullscreen) {
+		await element.requestFullscreen()
+	} else if (element.webkitRequestFullscreen) {
+		await element.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT)
+	} else if (element.mozRequestFullScreen) {
+		await element.mozRequestFullScreen()
+	} else if (element.msRequestFullscreen) {
+		await element.msRequestFullscreen()
+	}
```

Comment:
We only need `webkit` prefix for Safari now

https://caniuse.com/?search=requestFullscreen